components/inputs/input-text.js

```diff
@@ -444,27 +480,29 @@ class InputText extends FormElementMixin(SkeletonMixin(RtlMixin(LitElement))) {
 		this._hovered = false;
 	}
 
-	_onAfterSlotChange(e) {
-		const afterContent = e.target.assignedNodes({ flatten: true });
-		this._hasAfterContent = (afterContent && afterContent.length > 0);
-	}
-
-	_onSlotChange(e) {
```

Comment:
By measuring the container instead of the slotted content individually, we are able to avoid `getComputedStyle`, but it also makes the width calculation simpler, not just for the unit case, but also for the simpler case that was originally supported.